### PR TITLE
MULE-9566: Content-Type set in request builder is not considered

### PR DESCRIPTION
**Pull request by canoasan**

### CODE REVIEW

File: modules/http/src/main/java/org/mule/module/http/internal/request/MuleEventToHttpRequest.java [hunk 1]
```diff
@@ -74,7 +74,10 @@ public HttpRequestBuilder create(MuleEvent event, String resolvedMethod, String
 
         for (String outboundProperty : event.getMessage().getOutboundPropertyNames())
         {
-            builder.addHeader(outboundProperty, event.getMessage().getOutboundProperty(outboundProperty).toString());
+            if (!builder.getHeaders().containsKey(outboundProperty))
```
Comment (afelisatti) [inline modules/http/src/main/java/org/mule/module/http/internal/request/MuleEventToHttpRequest.java]: Wouldn't this make the 3.6.x branch different from 3.x? Outbound properties take precedence over the request builder AFAIK. Why is this needed?

Comment (canoasan) [inline modules/http/src/main/java/org/mule/module/http/internal/request/MuleEventToHttpRequest.java]: What about this specific scenario? The test is the same in both branches and in 3.6.x it fails if I let outbound properties replace the headers defined in the request builder.
